src/server/lib/okta/api/errors.ts

```diff
@@ -11,8 +11,33 @@ import {
   ResourceNotFoundError,
   ActivateUserFailedError,
   OperationForbiddenError,
+  AuthenticationFailedError,
+  TooManyRequestsError,
 } from '@/server/models/okta/Error';
 
+const extractErrorResponse = async (
+  response: Response,
+): Promise<ErrorResponse> => {
+  try {
+    return await response.json().then((json) => {
+      const error = json as ErrorResponse;
+      return {
+        errorCode: error.errorCode,
+        errorSummary: error.errorSummary,
+        errorLink: error.errorLink,
+        errorId: error.errorId,
+        errorCauses: error.errorCauses,
+      };
+    });
+  } catch (error) {
+    throw new OktaAPIResponseParsingError(`${error}`);
```

Comment:
good shout, i'll refactor the implementation of `OktaAPIResponseParsingError` to check the `typeof error` and handle that way rather than just casting to string

this will only be used in edge cases when the response from okta fails to get parsed

Reply:
sounds good